src/sql/magic_cmd.py

```diff
@@ -210,6 +211,81 @@ def execute(self, cmd_name="", others="", cell="", local_ns=None):
 
             return report
 
+        elif cmd_name == "snippets":
+            parser = CmdParser()
+            parser.add_argument(
+                "-d", "--delete", type=str, help="Delete stored snippet", required=False
+            )
+            parser.add_argument(
+                "-fd",
+                "--delete-force",
+                type=str,
+                help="Force Delete stored snippet",
+                required=False,
+            )
+            parser.add_argument(
+                "-da",
+                "--delete-force-all",
+                type=str,
+                help="Force Delete all stored snippets",
+                required=False,
+            )
+            args = parser.parse_args(others)
+            SNIPPET_ARGS = [args.delete, args.delete_force, args.delete_force_all]
+            if SNIPPET_ARGS.count(None) == len(SNIPPET_ARGS):
+                return ", ".join(util.get_all_keys())
+            if args.delete:
+                deps = util.get_key_dependents(args.delete)
+                if deps:
+                    deps = ", ".join(deps)
+                    raise exceptions.UsageError(
+                        f"The following tables are dependent on {args.delete}: {deps}."
+                        f"Pass --delete-force to only delete {args.delete}."
+                        f"Pass --delete-force-all to delete {deps} and {args.delete}"
+                    )
+                else:
+                    key = args.delete
+                    remaining_keys = util.del_saved_key(key)
+                    display_msg = f"Deleted snippet : {key}"
+                    if remaining_keys:
+                        display_msg = (
+                            f"{display_msg}. Current saved snippets : "
+                            f"{', '.join(remaining_keys)}"
+                        )
+                    else:
+                        display_msg = (
+                            f"{display_msg}. There are no more saved snippets."
+                        )
+                    return display_msg
+            elif args.delete_force:
+                key = args.delete_force
+                deps = util.get_key_dependents(key)
+                warnings.warn(
```

Comment:
this should not be a warning, but a print statement (it's a confirmation message)

there are two cases:

- if the deleted snippet didn't have any dependencies just say "{snippet} has been deleted." (past tense)
- if there are dependencies say: "{snippet} has been deleted. {a}, {b} depend on it."